tequilapi/endpoints/proposals.go

```diff
@@ -158,32 +157,15 @@ func noMetrics(p proposalRes) proposalRes { return p }
 func addMetrics(mc metrics.QualityOracle) func(p proposalRes) proposalRes {
 	connects := mc.ProposalsMetrics()
 	proposalsMetrics := make(map[string]json.RawMessage, len(connects))
-
-	var proposal struct {
-		ProposalID proposalRes `json:"proposalId"`
-	}
-	var metrics struct {
-		ConnectCount json.RawMessage `json:"connectCount"`
-	}
+	var proposal struct{ ProposalID proposalRes }
 
 	for _, c := range connects {
-		if err := json.Unmarshal(c, &proposal); err != nil {
-			log.Warn("Failed to parse proposal info")
-			return noMetrics
-		}
-
-		if err := json.Unmarshal(c, &metrics); err != nil {
-			log.Warn("Failed to parse metrics")
-			return noMetrics
-		}
-
-		out, err := json.Marshal(metrics)
+		json, err := metrics.Parse(c, &proposal)
```

Comment:
we should rename `connects` to `metrics` and `m` variable.

Reply:
Renamed to `receivedMetrics` since `metrics` conflicts with the imported package name.